### ISSUE TITLE
Web component: Content Block Card Video - Screen reader focus is stuck on close button. We are unable to navigate using Left/Right swipe.

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#5671]
## Detailed description

On Voiceover iPhone

Issue:
After closing the dialog visually it is disappeared. But, Screen reader focus is stuck on close button. We are unable to navigate using Left/Right swipe. This issue is not found in react Staging.

Expected:
After closing the dialog, User should be able to navigate to the parent page. And the dialog should disappear for the screen reader user as well.

**Comment by Praveen-sr** [comment]
Ram did verify the above bug, now its working fine.
But the dialog role is missing. and the focus is not trapped within the modal dialog.
cc @RobertaJHahn
